llvm/include/llvm/MC/CAS/MCCASObjectV1.h

```diff
@@ -413,9 +418,19 @@ class MCCASReader {
     return Target.isLittleEndian() ? support::little : support::big;
   }
 
+  Triple::ArchType getArch() { return Target.getArch(); }
+
   Expected<uint64_t> materializeGroup(cas::ObjectRef ID);
   Expected<uint64_t> materializeSection(cas::ObjectRef ID);
   Expected<uint64_t> materializeAtom(cas::ObjectRef ID);
+  Expected<uint64_t>
+  materializeDebugLineSection(const SmallVector<cas::ObjectRef> &Refs,
+                              StringRef RelocData);
+  Expected<SmallVector<char, 0>>
+  applyRelocationAddends(const SmallVector<cas::ObjectRef> &Refs,
+                         StringRef RelocData, StringLiteral KindString,
+                         StringRef SectionName, bool &IsPaddingRefPresent);
+  Expected<bool> isDebugLineSection(const SmallVector<cas::ObjectRef> &Refs);
```

Comment:
This doesn't need to be a member function if we make the argument a `SectionRef` instead.
And then we can place this function into a separate header.
And then we can re-use this function in the dump tool later!

Instead of the current call:
```
auto IsDebugLineSection = Reader.isDebugLineSection(*Refs);
```

we would just do:

```
auto IsDebugLineSection = isDebugLineSection(*this);
```

Reply:
actually, I would take this a step further and create a lightweight adaptor class, removing all magic offsets from the middle of the code:

```
DebugLineSectionAdaptor {
   ArrayRef<ObjectRef> Refs;

   DebugLineSectionAdaptor(ArrayRef<ObjectRef> Refs);
   Expected<ArrayRef<ObjectRef>> dropPaddingAndRelocationRefs();
   Expected<RelocationRef> getRelocationRef;
};
```